shared/src/main/scala/io/estatico/newtype/macros/NewTypeMacros.scala

```diff
@@ -141,18 +136,12 @@ private[macros] class NewTypeMacros(val c: blackbox.Context)
     val baseTypeDef = mkBaseTypeDef(clsDef, reprType, subtype)
     val typeTypeDef = mkTypeTypeDef(clsDef, tparamNames, subtype)
 
-    if (tparams.isEmpty) {
-      q"""
-        type $typeName = $objName.Type
-        trait $typesTraitName {
-          type Repr = $reprType
-          $baseTypeDef
-          trait Tag extends _root_.scala.Any
-          ${mkTypeTypeDef(clsDef, tparamNames, subtype)}
-        }
-        $newtypeObjDef
-      """
-    } else {
+    if (emitTrait) {
+      val newtypeObjParents = objParents :+ tq"$typesTraitName"
+      val newtypeObjDef = ModuleDef(
+        objMods, objName, Template(newtypeObjParents, objSelf, objDefs ++ companionExtraDefs)
```

Comment:
I wonder what was the reason to use `ModuleDef` directly instead of quasi-quotes but oh well, that's not the concern of this PR.

Reply:
I wondered that as well but didn't want to change too much with this :)